### PR TITLE
Add suggested rule changes RE dependencies

### PR DESCRIPTION
**Pull request by fsschneider**
Suggested by @danielsnider (Daniel Snider), with minimal edits by me (Frank Schneider).

@danielsnider I made some smaller edits, perhaps you can take a look to verify that you still agree with it.

### CODE REVIEW

File: RULES.md [hunk 1]
```diff
@@ -292,6 +302,24 @@ It is not allowed to compute any kind of pairwise metrics between the public wor
 - On a held-out workload, submissions are not allowed to find the nearest neighbor among the public workloads to set any hyperparameter.
 
 </details>
+<br>
+
+Valid submissions must rely on new algorithmic or mathematical ideas and should not use software engineering approaches to speed up basic operations in `PyTorch`, `JAX`, their dependencies, the operating system, or the hardware.
```
Comment (georgedahl) [inline RULES.md:307]: "basic" -> "primitive"

File: RULES.md [hunk 2]
```diff
@@ -292,6 +302,24 @@ It is not allowed to compute any kind of pairwise metrics between the public wor
...
+
+<details>
+<summary>Examples:</summary>
+
```
Comment (georgedahl) [inline RULES.md:311]: Add somewhere:
"In general, submissions can make clever, judicious, and efficient use of public APIs in JAX and/or PyTorch but should not be trying to optimize the internals of primitive operations and standard dependencies."

File: RULES.md [hunk 3]
```diff
@@ -292,6 +302,24 @@ It is not allowed to compute any kind of pairwise metrics between the public wor
...
+- Submitters are not allowed to have a load-balancing algorithm to vary the amount of work performed on the CPU, GPU, OS subsystems, or compute units such as Tensor cores.
+
+</details>
+
+##### Software Dependencies
+
+We require submissions to use specific versions of `PyTorch`/`JAX` as well as additional dependencies in order to facilitate fair comparisons. Submitters must build on top of these provided software packages, which might be provided as a docker container. Additional dependencies can be added as long as they include a comment describing what was added and why. Submitters are free to add dependencies that support new algorithmic and mathematical ideas but they should not circumvent the intention of the benchmark to measure training speedups due to new training methods. For example, software engineering techniques that lead to faster implementations of existing software, e.g. using newer versions of `PyTorch` or `JAX`, are not allowed and these are described in more detail in the [Disallowed submissions](#disallowed-submissions) Section. In case of doubts, these additional dependencies will be judged by the spirit jury.
```
Comment (georgedahl) [inline RULES.md:322]: "Docker" should be capitalized as a proper noun

**Review by georgedahl — APPROVED** [review]

**Review by danielsnider — APPROVED** [review]
@fsschneider, I approve both your modifications and the suggestions from @georgedahl.

**Pull request merged**